synapseclient/api/agent_services.py

```diff
@@ -0,0 +1,259 @@
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
+    Registers an agent with Synapse OR gets existing agent registration.
+
+    Arguments:
+        cloud_agent_id: The cloud provider ID of the agent to register.
+        cloud_alias_id: The cloud provider alias ID of the agent to register.
+                        In the Synapse API, this defaults to 'TSTALIASID'.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The registered agent matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistration.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Request matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistrationRequest.html>
+    request = {
+        "awsAgentId": cloud_agent_id,
+        "awsAliasId": cloud_alias_id if cloud_alias_id else None,
+    }
+    return await client.rest_put_async(
+        uri="/agent/registration", body=json.dumps(request)
+    )
+
+
+async def get_agent(
+    registration_id: str, synapse_client: Optional["Synapse"] = None
+) -> Dict[str, Any]:
+    """
+    Gets information about an existing agent registration.
+
+    Arguments:
+        registration_id: The ID of the agent registration to get.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The requested agent registration matching
+            <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/AgentRegistration.html>
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_get_async(uri=f"/agent/registration/{registration_id}")
+
+
+async def start_session(
+    access_level: str,
+    agent_registration_id: str,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """
+    Starts a new chat session with an agent.
+
+    Arguments:
+        access_level: The access level of the agent.
+        agent_registration_id: The ID of the agent registration to start the session for.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    # Request matching <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/agent/CreateAgentSessionRequest.html>
```

Comment:
Might make sense to put this in the docstring so it can be referenced  to understand what access_level is supposed to be